packages/path_provider/path_provider/lib/path_provider.dart

```diff
@@ -7,6 +7,7 @@ import 'dart:io' show Directory, Platform;
 
 import 'package:flutter/foundation.dart' show kIsWeb, visibleForTesting;
 import 'package:path_provider_linux/path_provider_linux.dart';
+import 'package:path_provider_windows/path_provider_windows.dart';
```

Comment:
@timsneath It looks like this linux import breaks the support on all platforms in 1.6.10: https://pub.dev/packages/path_provider#-analysis-tab-
As a result it also breaks all platform supports for packages that depend on path_provider.
I guess both the linux import and the windows import have to be conditional imports here.

Reply:
[Add Linxu support](https://github.com/flutter/plugins/pull/2789) seems ok?

According to https://github.com/flutter/flutter/issues/52267, it is a workround until flutter tool supports `dartPluginClass` on native plugins

Reply:
The reason I came to the PR is actually because the Linux support is not OK. 
Since that PR on the 2nd of June path_provider doesn't support any platform anymore according to pub.dev. (That's the version 1.6.10 I'm talking about in the comment.)

It wouldn't really be a problem if it is only this package, but every package that depends on path_provider has this error when they publish a new version. A large part of pub breaks for 2 platforms that aren't even in alpha stage yet.

2 issues I found that are possibly related, but I expect more issues to be submitted because of this: 
https://github.com/flutter/flutter/issues/58954
https://github.com/dart-lang/pub-dev/issues/3680

I also found that comment that this is a workaround indeed, but IMO it is a very bad work around for a platform in technical preview when it breaks platforms that are stable.

Reply:
> @timsneath It looks like this linux import breaks the support on all platforms in 1.6.10: https://pub.dev/packages/path_provider#-analysis-tab-
> As a result it also breaks all platform supports for packages that depend on path_provider.

Pana's analysis is wrong, leading to incorrect information being displayed on pub.dev; that's certainly not ideal, but is very different from "breaks all platform support". If you have examples of actual compilation or runtime failures resulting from this, please file a new bug.

> I guess both the linux import and the windows import have to be conditional imports here.

There is no conditional import that selects between individual platforms.

> flutter/flutter#58954

This is entirely unrelated; I've added a comment to that effect.

Let's continue any discussion of the workaround on https://github.com/flutter/flutter/issues/52267 (and/or https://github.com/dart-lang/pub-dev/issues/3680 for that specific side-effec), since the structure at issue here predates this PR.

Reply:
Indeed, it doesn't really break support, but as Flutter is getting more supported platforms the badges to show the platform support become more important.

Reply:
Thanks for the report, Rene!

Resolving this conversation, since as @stuartmorgan noted, it's entirely orthogonal to this pull request.